Entities/Scene.cpp

```diff
@@ -1616,10 +1624,14 @@ int Scene::Save(Writer &writer) const
 void Scene::Destroy(bool notInherited)
 {
     delete m_pTerrain;
-    delete m_pPathFinder;
+    
+    for (int i = 0; i < sc_pathfinderCount; ++i) {
```

Comment:
Would be nice to uses an array of unique_ptrs do you don't have to delete. I know Scene isn't a cleaned up file so it's using raw pointers, but where reasonably possible we try to swap to smort pointers.